# HG changeset patch
# User Jordi Gutiérrez Hermoso <jordigh@gmail.com>
# Date 1294302687 18000
#      Thu Jan 06 03:31:27 2011 -0500
# Node ID e1a5bf3ea1d97f6b9552c13931334a2c3cf9a447
# Parent  48cb431db87bf3aecf2570c8da3a0d0b72473860
Add a GUD mode option to run-octave.in

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+2011-01-06  Jordi Gutiérrez Hermoso  <jordigh@gmail.com>
+
+	* run-octave.in: Add a -gud option so that gdb can be called from
+	Emacs's GUD mode and similar.
+
 2011-01-06  John W. Eaton  <jwe@octave.org>
 
 	* mkoctfile.in, mkoctfile.cc.in: Substitute and use LAPACK_LIBS.
 	Bug #32009.
 
 2010-01-04  Rik  <octave@nomad.inbox5.com>
 
 	* README.MacOS: Update CFLAGS recommendation to use -O2.
diff --git a/run-octave.in b/run-octave.in
--- a/run-octave.in
+++ b/run-octave.in
@@ -45,20 +45,26 @@ d2_path=`echo "$d2_list" | $AWK '{ t = (
 d3_path=`echo "$d3_list" | $AWK '{ t = (s $0); s = t; } END { sub (/:$/, "", s); print s; }'`
 d4_path=`echo "$d4_list" | $AWK '{ t = (s $0); s = t; } END { sub (/:$/, "", s); print s; }'`
 
 LOADPATH="$d1_path:$d2_path:$d3_path:$d4_path"
 IMAGEPATH="$top_srcdir/scripts/image"
 DOCFILE="$builddir/doc/interpreter/doc-cache"
 INFOFILE="$top_srcdir/doc/interpreter/octave.info"
 
+## Checking for string equality below with prepended x's in order to
+## handle problems with empty strings.
 if [ $# -gt 0 ]; then
   if [ "x$1" = "x-g" ]; then
     driver="gdb --args"
     shift
+  elif [ "x$1" = "x-gud" ]; then
+    ## Frontends for gdb (e.g. Emacs's GUD mode) need --annotate=3
+    driver="gdb --annotate=3 --args"
+    shift
   elif [ "x$1" = "x-valgrind" ]; then
     driver="valgrind --tool=memcheck"
     shift
   elif [ "x$1" = "x-strace" ]; then
     driver="strace -o octave.trace"
     shift
   fi
 fi
